debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -195,6 +195,7 @@ private void stop(boolean restart) {
 
             if (restart && restartDelay == null) {
                 restartDelay = ElapsedTimeStrategy.constant(Clock.system(), 10_000);
+                restartDelay.hasElapsed();
```

Comment:
This doesn't do anything; some left-over?

Reply:
@gunnarmorling Please take a look into the called method and try the behaviour before and after. The first hasElpased only sets the target timeot time and immediately returns.

Thus the log contained message will restart after 10 seconds but restarted without any delay. By introducing this call the first restart really happens after the delay.

Reply:
Uf, that sounds nasty. Shouldn't that initialization be done upon construction of the `ElapsedTimeStrategy`. More importantly, though, this might explain the log message flood reported in [DBZ-1889](https://issues.redhat.com/browse/DBZ-1889). If *calling*  `hasElapsed()` resets the timer, it's not useful for the purpose I had in mind. I was expecting that the timer is independent of this method be called. I think this needs fixing before releasing.